qiskit/algorithms/evolvers/pvqd.py

```diff
@@ -0,0 +1,458 @@
+        initial_parameters: Optional[np.ndarray] = None,
+        timestep: Optional[float] = None,
+        optimizer: Optional[Union[Optimizer, Minimizer]] = None,
+        expectation: Optional[ExpectationBase] = None,
+        initial_guess: Optional[np.ndarray] = None,
+        evolution: Optional[EvolutionSynthesis] = None,
+        gradients: bool = True,
+        quantum_instance: Optional[Union[Backend, QuantumInstance]] = None,
+    ) -> None:
+        """
+        Args:
+            ansatz: A parameterized circuit preparing the variational ansatz to model the
+                time evolved quantum state.
+            initial_parameters: The initial parameters for the ansatz.
+            timestep: The time step.
+            optimizer: The classical optimizers used to minimize the overlap between
+                Trotterization and ansatz. Can be either a :class:`.Optimizer` or a callable
+                using the :class:`.Minimizer` protocol.
+            expectation: The expectation converter to evaluate expectation values.
+            initial_guess: The initial guess for the first VQE optimization. Afterwards the
+                previous iteration result is used as initial guess. If None, this is set to
+                a random vector with elements in the interval :math:`[-0.01, 0.01]`.
+            evolution: The evolution synthesis to use for the construction of the Trotter step.
+                Defaults to first-order Lie-Trotter decomposition.
+            gradients: If True, use the parameter shift rule to compute gradients. If False,
+                the optimizer will not be passed a gradient callable.
+            quantum_instance: The backend of quantum instance used to evaluate the circuits.
+        """
+        if evolution is None:
+            evolution = LieTrotter()
+
+        self.ansatz = ansatz
+        self.initial_parameters = initial_parameters
+        self.timestep = timestep
+        self.optimizer = optimizer
+        self.initial_guess = initial_guess
+        self.expectation = expectation
+        self.evolution = evolution
+        self.gradients = gradients
+
+        self._sampler = None
+        self.quantum_instance = quantum_instance
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Return the current quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Optional[Union[Backend, QuantumInstance]]) -> None:
+        """Set the quantum instance and circuit sampler."""
+        if quantum_instance is not None:
+            if not isinstance(quantum_instance, QuantumInstance):
+                quantum_instance = QuantumInstance(quantum_instance)
+            self._sampler = CircuitSampler(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+
+    def step(
+        self, hamiltonian: OperatorBase, theta: np.ndarray, dt: float, initial_guess: np.ndarray
```

Comment:
```suggestion
        self, hamiltonian: OperatorBase, theta: np.ndarray, dt: float, initial_point: np.ndarray
```
Would that be more suitable and also compliant with other algorithms?

Reply:
I see. Can we find a more descriptive yet still concise name? In my opinion, it is unclear what the guess refers to here.

Reply:
Added in 9c0f9ae19